apps/ta_dump.py

```diff
@@ -113,9 +116,132 @@ def adc_integral_hist(adc_integrals):
     plt.title("TA ADC Integral Histogram")
     plt.xlabel("ADC Integral")
 
+    plt.tight_layout()
     plt.savefig("adc_integral_histogram.svg")
     plt.close()
 
+def plot_time_window_summary(ta_data, tp_data, quiet=False):
+    """
+    Plot summary statistics on time windows.
+
+    This uses 2 definitions for time windows:
+        Direct Diff: max(tp.time_start) - ta.time_start,
+        Max Diff: max(tp.time_start + tp.time_over_threshold) - ta.time_start
+    """
+    direct_diff = []
+    max_diff = []
+
+    for ta_frag, tp_frag in zip(ta_data, tp_data):
+        for ta, tps in zip(ta_frag, tp_frag):
+            direct_diff += list(np.max(tps['time_start']) - ta['time_start'])
+            max_diff += list(np.max(tps['time_start'] + tps['time_over_threshold']) - ta['time_start'])
```

Comment:
I am wanting to make that change in the `TPData` and `TAData` classes. At the moment, they separate by fragments, but removing this separation would make it easier to run through.

Reply:
At the moment, I actually can't do the pre-allocation for either data format because of the fragment separation: it is unknown how many TAs/TPs are within a fragment unless they are iterated through. The number of TAs/TPs can be added as a data member to each of `TPData` and `TAData`, but I think that removing the fragment separation will be more useful while also solving this problem.